src/components/MovieDetails/MovieDetails.js

```diff
@@ -0,0 +1,99 @@
+import './MovieDetails.css';
+import backIcon from '../../images/back-icon.png';
+import { humanizeDate } from '../../helperFunctions';
+import { useEffect, useState } from 'react';
+import { checkServerError } from '../../helperFunctions';
+import PropTypes from 'prop-types';
+import { useParams, Link } from 'react-router-dom'
+import Loader from '../LoadSpinner/LoadSpinner';
+
+const MovieDetails = ({ goToHomeView, getData }) => {
+  const [details, setDetails] = useState({});
+  const [serverError, setServerError] = useState(false);
+  const [waitingForSingleFetch, setWaitingForSingleFetch] = useState(true);
+
+  const chosenID = useParams().id
+
+  useEffect(() => {
+    
+    getData(`movies/${chosenID}`)
+      .then(data => {
+        if (data.movie) {
+          const fetchedMovie = data.movie
+          setDetails(fetchedMovie);
+          setWaitingForSingleFetch(false)
+        } else {
+          setServerError(true);
+          setWaitingForSingleFetch(false);
+        }
+      })
+      .catch((err) => {
+        console.error(err);
+        setServerError(true);
+        setWaitingForSingleFetch(false);
+      })
+  }, [])
```

Comment:
Could prob just import getData instead of passing down the func. I would also add it to the apicalls file. Something like a method called `const getMovieById = () = {}`

Reply:
Right. Totally.